src/cljs/rems/administration/create_catalogue_item.cljs

```diff
@@ -141,78 +111,22 @@
 (rf/reg-event-fx ::create-catalogue-item create-catalogue-item!)
 (rf/reg-event-fx ::edit-catalogue-item edit-catalogue-item!)
 
-(defn- fetch-workflows []
-  (fetch "/api/workflows"
-         {:handler #(rf/dispatch [::fetch-workflows-result %])
-          :error-handler (flash-message/default-error-handler :top "Fetch workflows")}))
-
-(rf/reg-fx ::fetch-workflows fetch-workflows)
-
-(rf/reg-event-fx
- ::fetch-workflows-result
- (fn [{:keys [db]} [_ workflows]]
-   (-> (assoc db ::workflows workflows)
-       (update-loading))))
-
-(rf/reg-sub ::workflows (fn [db _] (::workflows db)))
-
-(defn- fetch-resources []
-  (fetch "/api/resources"
-         {:handler #(rf/dispatch [::fetch-resources-result %])
-          :error-handler (flash-message/default-error-handler :top "Fetch resources")}))
-
-(rf/reg-fx ::fetch-resources fetch-resources)
-
-(rf/reg-event-fx
- ::fetch-resources-result
- (fn [{:keys [db]} [_ resources]]
-   (-> (assoc db ::resources resources)
-       (update-loading))))
-
-(rf/reg-sub ::resources (fn [db _] (::resources db)))
-
-
-(defn- fetch-forms []
-  (fetch "/api/forms"
-         {:handler #(rf/dispatch [::fetch-forms-result %])
-          :error-handler (flash-message/default-error-handler :top "Fetch forms")}))
-
-(rf/reg-fx ::fetch-forms fetch-forms)
+(fetcher/reg-fetcher ::workflows "/api/workflows")
+(fetcher/reg-fetcher ::resources "/api/resources")
+(fetcher/reg-fetcher ::forms "/api/forms")
 
-(rf/reg-event-fx
- ::fetch-forms-result
- (fn [{:keys [db]} [_ forms]]
-   (-> (assoc db ::forms forms)
-       (update-loading))))
-
-(rf/reg-sub ::forms (fn [db _] (::forms db)))
+(defn- fetch-catalogue-item-success [db [_ {:keys [wfid resource-id formid localizations organization]}]]
+  (-> db
+      (assoc ::form {:workflow (item-by-id (::workflows db) :id wfid)
+                     :resource (item-by-id (::resources db) :id resource-id)
```

Comment:
are you sure `::resources` and `::workflows` are available already?

Reply:
Good question, I'm thinking that the assumption was there already in the previous code.

Reply:
I think it used to work because you moved to editing state after loading the page first, and only then caused the catalogue item to be loaded.

Reply:
Ah you're right. Not sure what to do about this. Add a TODO?

Reply:
Oh actually it did also have explicit tracking of the loading state for this.

Reply:
right, `(update-loading)`